GeneratorInterface/Pythia8Interface/plugins/JetMatchingEWKFxFx.cc

```diff
@@ -0,0 +1,998 @@
+  //  onto them, leaving a remnant of hard emissions.
+  for (int i = 0; i < nParton; ++i) {
+    scaleF = eventProcessOrig[0].e() / workEventJet[typeIdx[2][i]].pT();
+    tempEventJet[typeIdx[2][i]].rescale5(scaleF);
+  }
+
+  if (!hjSlowJet->setup(tempEventJet)) {
+    errorMsg(
+        "Warning in JetMatchingMadgraph:matchPartonsToJets"
+        "Heavy: the SlowJet algorithm failed on setup");
+    return NONE;
+  }
+
+  while (hjSlowJet->sizeAll() - hjSlowJet->sizeJet() > 0) {
+    if (hjSlowJet->dNext() > qCutSq)
+      break;
+    hjSlowJet->doStep();
+  }
+
+  int nCLjets(0);
+  // Count the number of clusters with pT>qCut.  This includes the
+  //  original hard partons plus any hard emissions.
+  for (int idx = 0; idx < hjSlowJet->sizeAll(); ++idx) {
+    if (hjSlowJet->pT(idx) > sqrt(qCutSq))
+      nCLjets++;
+  }
+
+  // Debug printout.
+  if (MATCHINGDEBUG)
+    hjSlowJet->list(true);
+
+  // Count of the number of hadronic jets in SlowJet accounting
+  //  int nCLjets = nClus - nJets;
+  // Get number of partons. Different for MLM and FxFx schemes.
+  int nRequested = nParton;
+
+  // Veto event if too few hadronic jets
+  if (nCLjets < nRequested) {
+    if (MATCHINGDEBUG)
+      cout << "veto : other LESS_JETS " << endl;
+    if (MATCHINGDEBUG)
+      cout << "nCLjets = " << nCLjets << "; nRequest = " << nRequested << endl;
+    return LESS_JETS;
+  }
+
+  // In exclusive mode, do not allow more hadronic jets than partons
+  if (exclusive) {
+    if (nCLjets > nRequested) {
+      if (MATCHINGDEBUG)
+        cout << "veto : excl other MORE_JETS" << endl;
+      return MORE_JETS;
+    }
+  }
+
+  // No extra jets were present so no veto
+  return NONE;
+}
+
+bool JetMatchingEWKFxFx::doShowerKtVeto(double pTfirst) {
+  // Only check veto in the shower-kT scheme.
+  if (!doShowerKt)
+    return false;
+
+  // Reset veto code
+  bool doVeto = false;
+
+  // Find the (kinematical) pT of the softest (light) parton in the hard
+  // process.
+  int nParton = typeIdx[0].size();
+  double pTminME = 1e10;
+  for (int i = 0; i < nParton; ++i)
+    pTminME = min(pTminME, eventProcess[typeIdx[0][i]].pT());
+
+  // Veto if the softest hard process parton is below Qcut.
+  if (nParton > 0 && pow(pTminME, 2) < qCutSq)
+    doVeto = true;
+
+  // For non-highest multiplicity, veto if the hardest emission is harder
+  // than Qcut.
+  if (exclusive && pow(pTfirst, 2) > qCutSq) {
+    doVeto = true;
+    // For highest multiplicity sample, veto if the hardest emission is harder
+    // than the hard process parton.
+  } else if (!exclusive && nParton > 0 && pTfirst > pTminME) {
+    doVeto = true;
+  }
+
+  // Return veto
+  return doVeto;
+}
+
+// Function to get the current number of partons in the Born state, as
+// read from LHE.
+
+int JetMatchingEWKFxFx::npNLO() {
+  string npIn = infoPtr->getEventAttribute("npNLO", true);
+  int np = (npIn.empty()) ? atoi((char*)npIn.c_str()) : -1;
```

Comment:
One doubt of the last minute: is it correct that if `npIn` is empty you compute `npIn.c_str()`? It does not look so...

In any case `c_str()` returns a `const char*`, which is exaclty the argument needed by `std::atoi`: why do you need to cast a (non const) `char*`?

I think this should be (but please check):
```suggestion
  int np = !npIn.empty() ? atoi(npIn.c_str()) : -1;
```